client/state/comments/selectors.js

```diff
@@ -72,21 +72,18 @@ export const getPostOldestCommentDate = createSelector(
  * @param {String} status String representing the comment status to show. Defaults to 'approved'.
  * @return {Object} comments tree, and in addition a children array
  */
-export const getPostCommentsTree = createSelector(
-	( state, siteId, postId, status = 'approved' ) => {
-		const allItems = getPostCommentItems( state, siteId, postId );
-		const items = status !== 'all' ? filter( allItems, { status } ) : allItems;
+export const getPostCommentsTree = ( state, siteId, postId, status = 'approved' ) => {
```

Comment:
I think you _really_ want this to be memoized. Otherwise you're regenerating the comment tree whenever anything in state changes and you have a comment component attached.